example/streaming/server.js

```diff
@@ -0,0 +1,169 @@
+});
+
+const footer = layout.client.register({
+    name: 'footer',
+    uri: 'http://localhost:6104/manifest.json',
+});
+
+layout.css({ value: '/css' });
+
+const app = express();
+
+app.use(layout.pathname(), layout.middleware());
+
+app.get('/foo/css', (req, res) => {
+    res.set('Content-Type', 'text/css');
+    res.send(`
+        @keyframes pulse {
+            0% {
+                background-color: #e0e0e0;
+            }
+            50% {
+                background-color: #f0f0f0;
+            }
+            100% {
+                background-color: #e0e0e0;
+            }
+        }
+        .skeleton {
+            width: 100%;
+            background-color: #e0e0e0;
+            border-radius: 5px;
+            animation: pulse 1.5s infinite ease-in-out;
+            margin: 0;
+			margin-bottom: 20px;
+			box-sizing: border-box;
+        }
+        .skeleton.header {
+            height:79px;
+        }
+        .skeleton.menu {
+            height:40px;
+        }
+        .skeleton.content {
+            height:60px;
+        }
+        .skeleton.footer {
+            height:60px;
+        }
+	`);
+});
+
+app.get(layout.pathname(), async (req, res) => {
+    const incoming = res.locals.podium;
+
+    incoming.view = {
+        title: 'Example streaming application',
+    };
+
+    const headerFetch = header.fetch(incoming);
+    const menuFetch = menu.fetch(incoming);
+    const contentFetch = content.fetch(incoming);
+    const footerFetch = footer.fetch(incoming);
+
+    incoming.hints.on('complete', async ({ js, css }) => {
+        // set the assets on httpincoming so that they are available in the document template
+        incoming.js = [...incoming.js, ...js];
+        incoming.css = [...incoming.css, ...css];
+
+        // set up the stream which will send the document template head
+        const stream = res.podiumStream();
+
+        // stream in the document body with slot placeholders for podlets
+        stream.send(`
+            <template shadowrootmode="open">
+                <link href="/foo/css" type="text/css" rel="stylesheet">
+                <div class="container">
+                    <div>
+                        <div>
+                            <slot name="header"><div class="skeleton header"></div></slot>
+                        </div>
+                    </div>
+                    <div>
+                        <div>
+                            <slot name="menu"><div class="skeleton menu"></div></slot>
+                        </div>
+                        <div>
+                            <slot name="content"><div class="skeleton content"></div></slot>
+                        </div>
+                    </div>
+                    <div>
+                        <div>
+                            <slot name="footer"><div class="skeleton footer"></div></slot>
+                        </div>
+                    </div>
+                </div>
+            </template>
+        `);
```

Comment:
In this case we are sending a DSD template with slots and placeholder skeleton html so that we can replace these by streaming in podlet content later.